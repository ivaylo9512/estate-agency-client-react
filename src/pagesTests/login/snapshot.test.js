import React from 'react';
import Login from '../../pages/login';
import { shallow } from 'enzyme';
import * as redux from 'react-redux';

describe("Login snapshot tests", () => {
    let selectorSpy;

    beforeAll(() => {
        selectorSpy = jest.spyOn(redux, 'useSelector');
    
        const spyOnUseDispatch = jest.spyOn(redux, 'useDispatch');
        spyOnUseDispatch.mockReturnValue(jest.fn());
    });

    const createWrapper = (state) => {
        selectorSpy.mockReturnValue(state);
        return shallow(
            <Login />
        )
    }

    it('renders correctly', () => {
        const wrapper = createWrapper({ isLoading: false, error: null });
     
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with passed errors props', () => {
        const wrapper = createWrapper({ isLoading: false, error: 'Bad credentials.' });

        expect(wrapper).toMatchSnapshot();
    });
});