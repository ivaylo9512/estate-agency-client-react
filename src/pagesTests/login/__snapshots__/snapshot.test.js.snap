// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot tests renders correctly 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <input
      name="username"
      onChange={[Function]}
      placeholder="username or email"
      type=""
      value=""
    />
    <usePasswordInput__Container>
      <input
        autoComplete="current-password"
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
      <usePasswordInput__Button
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f06e",
                "M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z",
              ],
              "iconName": "eye",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </usePasswordInput__Button>
    </usePasswordInput__Container>
    <span>
      Don't have an account?
      <Link
        href="/register"
      >
         Sign up.
      </Link>
    </span>
    <button>
      login
    </button>
  </form>
</section>
`;

exports[`Login snapshot tests renders correctly with passed errors props and page 0 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <input
      name="username"
      onChange={[Function]}
      placeholder="username or email"
      type=""
      value=""
    />
    <usePasswordInput__Container>
      <input
        autoComplete="current-password"
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
      <usePasswordInput__Button
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f06e",
                "M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z",
              ],
              "iconName": "eye",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </usePasswordInput__Button>
    </usePasswordInput__Container>
    <div
      className="errors"
    >
      Bad credentials.
    </div>
    <span>
      Don't have an account?
      <Link
        href="/register"
      >
         Sign up.
      </Link>
    </span>
    <button>
      login
    </button>
  </form>
</section>
`;
